src/connector/src/sink/formatter/mod.rs

```diff
@@ -50,3 +54,68 @@ macro_rules! tri {
         }
     };
 }
+
+#[expect(clippy::enum_variant_names)]
+pub enum SinkFormatterImpl {
```

Comment:
I was thinking each sink would have their own enum, for example kinesis cannot be combined with a format that outputs binary key, or nats should not be combined with a format that outputs any (non-`()`) key. Neither of these is a big issue right now. And the only byproduct of sharing a single enum seems meaningful: debezium is supported for pulsar. We can figure out how to use different enums when the need does come later.